strategies/sendfile.macos.scm

```diff
@@ -7,11 +7,16 @@
 ;; EAGAIN may be signaled even when partial data is sent, but the caller expects EAGAIN
 ;; to mean zero bytes sent, so we return the number of bytes sent when non-zero.
 (define %sendfile-implementation
-  (foreign-lambda* double ((integer src) (integer dst) (double offset) (unsigned-integer to_send))
-    "off_t res = (off_t)to_send;"
-    "if(sendfile(src,dst,(off_t)offset,&res,NULL/*&hdtr*/,0) < 0){"
-    "  if(errno == EAGAIN)"
-    "    C_return(res == 0 ? -2 : (double)(offset + res));"
-    "  else C_return(-1);"
-    "}"
-    "C_return((double)(offset + res));"))
+  (foreign-lambda* ssize_t ((integer src) (integer dst) (unsigned-long offset) (unsigned-long to_send))
+    "
+    off_t res = to_send;
+    if(sendfile(src,dst,offset,&res,NULL,0) < 0) {
+      if(errno == EAGAIN) {
```

Comment:
Perhaps this should read `(errno == EGAIN || errno == EINTR)` like on the other platforms. The manpage (see link above) mentions that it can return this.

It might even be worth fixing that on the current `master` before this refactor is merged so we can fix it in C4 as well.

Reply:
Done in #6